### ISSUE TITLE
ImageConverter not found in .Net Standard lib referenced by .Net Framework 4.7.2 app

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/runtime#103702]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/ImageConverter-not-found-in-Net-Standar/10678473)._

---
\[severity:It bothers me. A fix would be nice\]  
I believe I may have found a bug. I’ve attached a sample app that demonstrates the issue better than my ability to explain it. Running a .Net Framework 4.7.2 app that references a .Net Standard 2.0 library. Both projects must reference a package that includes System.Drawing.Common as a transient dependency. When code within the .Net Standard 2.0 library attempts to initialize a System.Drawing.ImageConverter, the following exception is thrown:

`System.TypeLoadException: 'Could not load type 'System.Drawing.ImageConverter' from assembly 'System.Drawing.Common, Version=7.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.'`

Note that the app looks for ImageConverter within System.Drawing.Common. In this configuration, ImageConverter is actually located within the package System.Drawing. When I use reflection to load the System.Drawing assembly and retrieve an instance of ImageConverter from there, it succeeds.

The attached project should demonstrate this. Please note that in this case, the package EvoHtmlToPdf\_NetCore has the dependency on System.Drawing.Common and removing this package is not an option. EvoHtmlToPdf\_NetCore is used in the production application where this issue was discovered.

[ImageConverterTest.zip](https://aka.ms/dc/file?name=B99dc5a23c9c94871b3702f3d267b8950638533705511371995_ImageConverterTest.zip&tid=99dc5a23c9c94871b3702f3d267b8950638533705511371995)

---
#### Feedback Bot on 6/10/2024, 08:04 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by ericstj** [comment]
Confirmed that this problem exists in the latest 8.0.0 package as well.  I'm surprised APICompat doesn't catch it.

Seems like the types were moved down:
https://github.com/dotnet/runtime/commit/1dbdb8b78738a6da37ac2ea1937b1e16aef7937a

But at the time, we weren't building `netstandard2.0` at all - we were harvesting the reference.   Then `netstandard2.0` was added back, including these converters and for some reason the type forwards for .NETFramework never made it in.  The same bug exists in 6.0.0 and 8.0.0.  

We should fix this in latest (in winforms repo) and consider backport to 6.0.  Keeping this here for a moment before transferring so that I can determine why it was missed in 6.0.

**Comment by ericstj** [comment]
APICompat does find this problem:
``` 
apicompat package c:\Users\ericstj\.nuget\packages\system.drawing.common\8.0.0\system.drawing.common.8.0.0.nupkg --package-assembly-references ".NETFramework,Version=v4.6.2|C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6.2,C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6.2\facades" --package-assembly-references ".NETStandard,Version=v2.0|C:\Users\ericstj\.nuget\packages\netstandard.library\2.0.3\build\netstandard2.0\ref"  --package-assembly-references ".NETCoreApp,Version=v6.0|C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\6.0.31\ref\net6.0" --package-assembly-references ".NETCoreApp,Version=v7.0|C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\7.0.20\ref\net7.0" --package-assembly-references ".NETCoreApp,Version=v8.0|C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\8.0.6\ref\net8.0"
API compatibility errors between 'lib/netstandard2.0/System.Drawing.Common.dll' (left) and 'lib/net462/System.Drawing.Common.dll' (right):
CP0001: Type 'System.Drawing.FontConverter' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0001: Type 'System.Drawing.IconConverter' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0001: Type 'System.Drawing.ImageConverter' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0001: Type 'System.Drawing.ImageFormatConverter' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0002: Member 'System.Drawing.Imaging.Encoder System.Drawing.Imaging.Encoder.ColorSpace' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0002: Member 'System.Drawing.Imaging.Encoder System.Drawing.Imaging.Encoder.ImageItems' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0002: Member 'System.Drawing.Imaging.Encoder System.Drawing.Imaging.Encoder.SaveAsCmyk' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0002: Member 'System.Drawing.Imaging.EncoderParameterValueType.ValueTypePointer' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
CP0001: Type 'System.Drawing.Printing.MarginsConverter' exists on lib/netstandard2.0/System.Drawing.Common.dll but not on lib/net462/System.Drawing.Common.dll
API breaking changes found. If those are intentional, the APICompat suppression file can be updated by specifying the '--generate-suppression-file' parameter.
```
And indeed they're suppressed incorrectly in the latest codebase: https://github.com/dotnet/winforms/blob/346ceee33534e8f709e486ec1fbd05cc6ac9c689/src/System.Drawing.Common/src/CompatibilitySuppressions.xml#L5-L33

I think this was a case of clashing changes in 6.0.  We converted generated facades to authored type-forwards around the same time an API change happened to System.Drawing to add types to the netstandard2.0 surface.  As a result we missed adding these types to the net4x facade.

**Comment by ericstj** [comment]
cc @dhuitorigami who may be the same Darrin Huit who opened this feedback ticket.  Thank your for the excellent bug report.  The fix will be present in the 9.0 release of System.Drawing.Common.

**Comment by dhuitorigami** [comment]
Yep, the same. Thank you! :)
